src/serde/de.rs

```diff
@@ -19,20 +18,41 @@ where
     T: DeserializeOwned,
     I: IonInput,
 {
-    let mut reader = Reader::new(AnyEncoding, input)?;
-    let value = reader.expect_next()?;
-    let value_deserializer = ValueDeserializer::new(&value);
-    T::deserialize(value_deserializer)
+    let mut reader = SystemReader::new(AnyEncoding, input);
+    let item = reader.next_item()?;
+    match item {
+        SystemStreamItem::VersionMarker(marker) => {
+            // Note that this uses an older encoding here, but the format (i.e text or binary) stays the same which is why using the older encoding here works fine.
```

Comment:
```suggestion
            // Note that this uses the Ion version with which the IVM was encoded rather than
            // the Ion version the stream is switching to. We can do this because the format 
            // (i.e text or binary) stays the same when the version changes.
```